### PR TITLE
Use name param to set the cluster resource name

### PR DESCRIPTION
**Pull request by pivotal-nader-ziada**
Fixes: #431 

### Proposed changes: 

Get the name of the `ClusterResource` from a `name` param instead of the resource name to allow users to be more flexible with names. Using the resource metadata name forces users to only use DNS-1123 subdomain compliant names

### CODE REVIEW

File: docs/resources.md [hunk 1]
```diff
@@ -172,8 +172,8 @@ cluster. The kubeconfig will be placed in
 
 The Cluster resource has the following parameters:
 
-- Name: The name of the Resource is also given to cluster, will be used in the
-  kubeconfig and also as part of the path to the kubeconfig file
+- Name (required): The name to be given to the target cluster, will be used
```
Comment (shashwathi) [inline docs/resources.md]: If `Name` is required then can it adds value to add this check in resource validation.

**Comment by shashwathi** [comment]
/hold 

Adding hold for above changes. @pivotal-nader-ziada feel free to remove if you have addressed the comment

File: docs/resources.md (hunk 1 above)
Comment (pivotal-nader-ziada) [reply docs/resources.md]: added validation

**Pull request merged**